### PR TITLE
persist,dataflow: fix various issues in persisted source

### PR DESCRIPTION
**Pull request by aljoscha**
Early-access PR!

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: src/dataflow/src/source/mod.rs [hunk 1]
```diff
@@ -1500,9 +1500,14 @@ fn handle_message<S: SourceReader>(
...
-    let ts_cap = cap.delayed(&ts);
 
-    output.session(&ts_cap).give(Ok(SourceOutput::new(
+    // NOTE: It is **very** important that we emit with the capability that we are given. This
+    // capability is also used to emit updates about the current read offset, and we need to make
+    // sure that these updates happen at the same timestamp, such that we can have a consistent
+    // state if/when the source restarts from persisted offsets/data.
+    //
+    // TODO: This futzes with the timestamp that emitted data has, not sure that's a problem.
+    output.session(&cap).give(Ok(SourceOutput::new(
```
Comment (ruchirK) [inline src/dataflow/src/source/mod.rs:1516]: I think if I understand this issue:

we're emitting bindings that look like ((source offset, assigned time), capability time, diff)

and we're minting data that looks like (data, assigned time,diff)

and as you can see the times on the 2 are different

Comment (ruchirK) [reply src/dataflow/src/source/mod.rs:1516]: the change fixes this by emitting data at capability time - but we could go the other way and emit bindings at assigned time

Comment (ruchirK) [reply src/dataflow/src/source/mod.rs:1516]: maybe

**Comment by aljoscha** [comment]
Rebased on the big `mz_` rename PR.

**Comment by philip-stoev** [comment]
Item No 1. In this branch unfortunately the test case from #10756 returns more rows in the materialized view than there were rows in the source to begin with:

```
> SELECT COUNT(*) FROM a_view;
rows didn't match; sleeping to see if dataflow catches up 50ms 75ms 112ms 169ms 253ms 380ms 570ms 854ms 1s 2s 2s
^^^ +++
gh10756.td:29:1: error: non-matching rows: expected:
[["1000000"]]
got:
[["1000032"]]
Diff:
row missing: ["1000000"]
extra row: ["1000032"]
```

This failure is specific to this branch, does not happen on `main`.

**Pull request closed without merging**